integration_tests/src/janus.rs

```diff
@@ -56,3 +85,191 @@ impl<'a> Janus<'a> {
             .get_host_port_ipv4(Aggregator::INTERNAL_SERVING_PORT)
     }
 }
+
+/// Represents a running Janus test instance in this process.
+pub struct JanusInProcess {
+    socket_address: SocketAddr,
+    stopper: Stopper,
+    _ephemeral_datastore: EphemeralDatastore,
+}
+
+impl JanusInProcess {
+    /// Start a new Janus instance in the current process, using a separate ephemeral database,
+    /// configured to service the given task.
+    pub async fn new(task: &Task, role: Role) -> Self {
+        // Set up common utilities.
+        let stopper = Stopper::new();
+        let clock = RealClock::default();
+        let ephemeral_datastore = ephemeral_datastore().await;
+        let datastore = ephemeral_datastore.datastore(clock).await;
+        let encoded_datastore_key =
+            URL_SAFE_NO_PAD.encode(ephemeral_datastore.datastore_key_bytes());
+        let datastore_keys = Vec::from([encoded_datastore_key]);
+        let database_url = ephemeral_datastore
+            .connection_string()
+            .parse()
+            .expect("error parsing database URL");
+
+        // Provision the task.
+        datastore
+            .put_aggregator_task(&task.view_for_role(role).expect("invalid role"))
+            .await
+            .expect("task provisioning failed");
+
+        // Construct configuration and options for each component.
+        let common_binary_options = CommonBinaryOptions {
+            config_file: PathBuf::new(),
+            database_password: None,
+            datastore_keys,
+            otlp_tracing_metadata: Vec::new(),
+            otlp_metrics_metadata: Vec::new(),
```

Comment:
nit: could we provide `Default` on `CommonBinaryOptions` and do `CommonBinaryOptions { datastore_keys, ..Default::default() }`?